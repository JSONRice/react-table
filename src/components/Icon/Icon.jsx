import React from "react";
import styled from '@emotion/styled'
import PropTypes from "prop-types";

const StyledIcon = styled.span`
  display: flex;
  cursor: ${props => (props.pointerCursor ? `pointer` : `default`)};
  flex-direction: ${props => props.flexDirection};
  justify-items: center;
  align-items: center;

  ::before {
    display: flex;
    font-size: 14px;
    color: ${props => props.color};
    content: '${props => props.content}';
    transform: ${props => props.expand && `rotate(${props.rotateDeg}deg)`};
  }
`;

export const Icon = ({
  children,
  color,
  content,
  expand,
  flexDirection,
  fontFamily,
  fontSize,
  hideChildren,
  name,
  onClick,
  pointerCursor,
  rotateDeg,
  className,
  transparent
}) => {
  return (
    <StyledIcon
      className={className}
      color={color}
      content={content}
      expand={expand}
      flexDirection={flexDirection}
      fontSize={fontSize}
      fontFamily={fontFamily}
      name={name}
      onClick={onClick}
      pointerCursor={pointerCursor}
      rotateDeg={rotateDeg}
      transparent={transparent}
    >
      <>{!hideChildren && children}</>
    </StyledIcon>
  );
};

Icon.propTypes = {
  color: PropTypes.string,
  content: PropTypes.string,
  expand: PropTypes.bool,
  flexDirection: PropTypes.string,
  fontFamily: PropTypes.string,
  fontSize: PropTypes.number,
  hideChildren: PropTypes.bool,
  name: PropTypes.string,
  onClick: PropTypes.func,
  pointerCursor: PropTypes.bool,
  rotateDeg: PropTypes.number,
  className: PropTypes.string,
  transparent: PropTypes.bool
};

Icon.defaultProps = {
  color: 'black',
  content: '>',
  expand: false,
  flexDirection: "row",
  fontSize: 10,
  hideChildren: false,
  pointerCursor: true,
  rotateDeg: 90,
  transparent: false
};
