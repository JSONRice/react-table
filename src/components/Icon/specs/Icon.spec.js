import React from "react";
import renderer from "react-test-renderer";
import { Icon } from "../Icon";

describe("Icon", () => {
  const component = (
    <Icon
      name="triangleRight"
      toggleChildren={true}
      expand={true}
      onClick={() => {
        // noop
      }}
    >
      Here is a right triangle with toggle-able content.
    </Icon>
  );

  /**
   * On the first run of this test, Jest will generate a snapshot file automatically.
   * Run the following to update the snapshot (see package.json): npm run snapshot
   */
  it("renders successfully and matches snapshot", () => {
    const mounted = renderer.create(component).toJSON();
    expect(mounted).toMatchSnapshot();
  });
});
