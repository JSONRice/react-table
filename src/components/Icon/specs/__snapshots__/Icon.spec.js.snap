// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders successfully and matches snapshot 1`] = `
<span
  className="css-rb7dt2"
  color="black"
  fontSize={10}
  name="triangleRight"
  onClick={[Function]}
>
  Here is a right triangle with toggle-able content.
</span>
`;
